rfcs/012-merge-rebase-strategy.md

```diff
@@ -0,0 +1,81 @@
+
+## Best Practice
+
+There's no best practice on this matter but different workflows that are suitable for different teams.
+For Open Collective projects only core team members are merging. It's hard or probably impossible to enforce all the 
+community to interactively rebase Git history. Hence, currently we are mostly relying on squashing and merging PRs which are not 
+interactively rebased and have multiple commits, or those which have merge commits from main branch. 
+
+Whether contributors use interactive re-basing, or if they choose to use simple merging, a cleaner Git history could still be 
+achieved if PRs are squashed before merging. 
+
+Some major projects such as NextJS, React and Babel seem to use the same approach in maintaining their Git histories. Some examples are given below.
+
+For NextJS; https://github.com/vercel/next.js/pull/25198. In this case the contributor uses lot of commit messages and also merge from `canary` (the default branch of NextJS) to their branch. 
+However, the final result is just a single commit without any merges.
+
+![image](https://user-images.githubusercontent.com/12435965/122803708-d158cc00-d27b-11eb-8f99-ac6d7a26fa43.png)
+
+Similarly, here's a recent example for React; https://github.com/facebook/react/pull/21648. This seems a squash merge;
+
+![image](https://user-images.githubusercontent.com/12435965/122802824-b20d6f00-d27a-11eb-8db9-10ba529bb9a6.png)
+
+For Babel: https://github.com/babel/babel/pull/13419. Again this seems to be a squash merge;
+
+![image](https://user-images.githubusercontent.com/12435965/122803003-f39e1a00-d27a-11eb-8f00-f34d480af450.png)
+
+Therefore, it could be argued that the squash and merge approach is suitable and works in maintaining a linear Git history while
+also making sure the contributors get full flexibility to choose how they arrange their branch commits.
+
+### Downside of this approach
+
+One downside is that when squashing and merging the person who merge needs to verify the commit title and description and make sure
+it makes sense. The GitHub interface automatically creates a commit description by concatenating all the commits in the branch. 
+Although this seems to work in most cases it's probably a good practice to see if the commit(s) make sense while merging. 
+
+## Solution
+
+[Disable `Create a merge commit` from the GitHub settings](https://docs.github.com/en/github/administering-a-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests) 
+to help maintain a linear Git history. Additionally, suggest also disabling `Rebase and merge` as well although it could be argued to leave it in case someone would like to explicitly include
+several commits to the `main` branch. The squash and merge option should work for us in all cases. 
+
+As an alternative or a stricter enforcement of this standard we can also add the [`Require linear history` branch protection rule 
```

Comment:
An RFC should not have uncertainty on the outcome. Are we gonna do this or not, let's be straight about it.

Reply:
@znarf : Done, I've added lines to make sure we disable only the merge option and keep the squash and rebase options. Also we will enable the branch protection rule on the main branch to preserve linear history (no merge commits on main branch). Let me know if you see anything else. 😄